lib/ansible/modules/unarchive.py

```diff
@@ -788,7 +788,7 @@ def files_in_archive(self):
         locale = get_best_parsable_locale(self.module)
         rc, out, err = self.module.run_command(cmd, cwd=self.b_dest, environ_update=dict(LANG=locale, LC_ALL=locale, LC_MESSAGES=locale))
         if rc != 0:
-            raise UnarchiveError('Unable to list files in the archive')
+            raise UnarchiveError(err)
```

Comment:
In case the err doesn't indicate what was attempted, you could just add the stderr to the end:
```suggestion
            raise UnarchiveError('Unable to list files in the archive: %s' % err)
```
I see this message multiple places in this file that don't include the error - ideally these would be consistent.